src/rollup/RollupCreator.sol

```diff
@@ -60,20 +65,23 @@ contract RollupCreator is Ownable {
         IRollupEventInbox rollupEventInbox;
         IOutbox outbox;
         RollupProxy rollup;
+        bytes32 rollupSalt;
     }
 
     // After this setup:
     // Rollup should be the owner of bridge
     // RollupOwner should be the owner of Rollup's ProxyAdmin
     // RollupOwner should be the owner of Rollup
     // Bridge should have a single inbox and outbox
-    function createRollup(Config memory config, address expectedRollupAddr)
-        external
-        returns (address)
-    {
+    function createRollup(Config memory config) external returns (address) {
         CreateRollupFrame memory frame;
         frame.admin = new ProxyAdmin();
 
+        frame.rollupSalt = bytes32(deployedRollups++);
```

Comment:
should we just let people to pick a salt instead? this seems to same as using a nonce which can still cause the same issue

Reply:
Yeh, We might also want to hash the config and contract dependencies into the salt as well to ensure it couldnt be frontrun with different params?

Reply:
Hmm, the part I was trying to achieve is that you don't need to know the rollup address ahead of time, and you can instead just query the receipt logs to get that. Is there another use case where a user would want to know the rollup address ahead of time? We could perhaps make the salt a hash of their address and a nonce they specify.